.github/workflows/helm.yml

```diff
@@ -0,0 +1,162 @@
+          cache-from: type=gha,scope=${{ inputs.image }}
+          pull: true
+          build-args: |
+            NJS_DIR=internal/mode/static/nginx/modules/src
+            NGINX_CONF_DIR=internal/mode/static/nginx/conf
+            BUILD_AGENT=gha
+
+      - name: Install cloud-provider-kind
+        run: go install sigs.k8s.io/cloud-provider-kind@latest
+
+      - name: Run cloud-provider-kind
+        run: ~/go/bin/cloud-provider-kind & > cloud-provider-kind.log 2>&1
+
+      - name: Deploy Kubernetes
+        id: k8s
+        run: |
+          kind create cluster --name ${{ github.run_id }} --image=kindest/node:${{ inputs.k8s-version }}
+          kind load docker-image ${{ join(fromJSON(steps.ngf-meta.outputs.json).tags, ' ') }} ${{ join(fromJSON(steps.nginx-meta.outputs.json).tags, ' ') }} --name ${{ github.run_id }}
+          kubectl kustomize config/crd/gateway-api/standard | kubectl apply -f -
+
+      - name: Set up Python
+        uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d # v5.1.0
+        with:
+          python-version: "3.x"
+          check-latest: true
+
+      - name: Set up chart-testing
+        uses: helm/chart-testing-action@e6669bcd63d7cb57cb4380c33043eebe5d111992 # v2.6.1
+
+      - name: Install Chart
+        run: |
+          ct install --config .ct.yaml --helm-extra-set-args="--set=nginxGateway.image.tag=${{ steps.ngf-meta.outputs.version }} \
+          --set=nginx.image.repository=ghcr.io/nginxinc/nginx-gateway-fabric/nginx${{ inputs.image == 'plus' && '-plus' || ''}} \
+          --set=nginx.plus=${{ inputs.image == 'plus' }} \
+          --set=nginx.image.tag=${{ steps.nginx-meta.outputs.version }} \
+          --set=nginxGateway.image.pullPolicy=Never \
+          --set=nginx.image.pullPolicy=Never \
+          --set=nginxGateway.productTelemetry.enable=false"
+
+  helm-test-remote:
+    name: Helm Tests Remote
+    runs-on: ubuntu-22.04
+    # if: ${{ github.event_name == 'schedule' }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
+        with:
+          fetch-depth: 0
+
+      - name: Install cloud-provider-kind
+        run: go install sigs.k8s.io/cloud-provider-kind@latest
+
+      - name: Run cloud-provider-kind
+        run: ~/go/bin/cloud-provider-kind & > cloud-provider-kind.log 2>&1
+
+      - name: Deploy Kubernetes
+        id: k8s
+        run: |
+          kind create cluster --name ${{ github.run_id }} --image=kindest/node:${{ inputs.k8s-version }}
+          kubectl kustomize config/crd/gateway-api/standard | kubectl apply -f -
+          kubectl create namespace nginx-gateway
+
+      - name: Create k8s secret
+        if: ${{ inputs.image == 'plus' }}
+        run: |
+          kubectl create secret docker-registry nginx-plus-registry-secret --docker-server=private-registry.nginx.com --docker-username=${{ secrets.JWT_PLUS_REGISTRY }} --docker-password=none -n nginx-gateway
```

Comment:
Does this token need to be manually rotated? Feel free to chat offline if we can't share the details here.

Reply:
Unfortunately it's not avoidable at the moment

Reply:
Ultimately the real point of the test is to ensure we could pull an image from a private registry. Could kind be instrumented with a private registry that the image built in the pipeline could live in, that way we create the secret to pull from there?

Reply:
I'm not sure how we would set that up, where would the registry be hosted?

I think we need to make a plan to set up secrets rotation because we use the same secret (in a different format) for building the NGINX Plus images and we have others too.

Reply:
That's true.

I was just thinking of a local docker registry running on localhost.

Reply:
Oh it's easy to do that in the workflow then (we do it in the build workflow), the hard part is setting up the authentication if it's even possible 😅

Reply:
yeah that's the part I wasn't sure about. Some dummy username/password.

Reply:
@sjberman are you okay with trying this for now?

Reply:
Yes, though let's maybe create a ticket to explore the other option.